syntax = "proto3";

package streamingserver;

import "google/api/annotations.proto";

import "api/protos/req_type_view.proto";

// StreamingServerService defines the service that manages distributed clients
// for a given multiplexed audio channel stream.
service StreamingServerService {
    // Get represents the single endpoint used by any client type to obtain
    // any resource stored by or managed in the server.
    rpc Get(GetRequest) returns (GetResponse) {
        option (google.api.http) = {
            get: "/v1/get"
        };
    };

}

// GetRequest defines a request for any supported resource type that a client
// may fetch from the server.
message GetRequest {
    oneof req_type {
        // view_request returns visual data used by a client to render info
        ViewRequestType view_request = 1;
    }
}

// GetResponse defines a supported resource requested by a client.
// It may be one of (though not limited to) the following: (view,
// sound_group_membership, sound_group_info).
message GetResponse {
    
}